backend/parser/listener/listener.go

```diff
@@ -62,13 +93,27 @@ func (s *searchListener[T]) ExitAnd_col_expr(ctx *parser.And_col_exprContext) {
 	rules := s.rules[len(s.rules)-2:]
 	s.rules = s.rules[:len(s.rules)-2]
 	s.rules = append(s.rules, s.sb.And(rules...))
+
+	ops := s.ops[len(s.ops)-2:]
+	s.ops = s.ops[:len(s.ops)-2]
+	s.ops = append(s.ops, &FilterOperation{
+		Operator: OperatorAnd,
+		Filters:  Filters{ops[0], ops[1]},
+	})
 }
 
 func (s *searchListener[T]) EnterOr_col_expr(ctx *parser.Or_col_exprContext) {}
 func (s *searchListener[T]) ExitOr_col_expr(ctx *parser.Or_col_exprContext) {
 	rules := s.rules[len(s.rules)-2:]
 	s.rules = s.rules[:len(s.rules)-2]
 	s.rules = append(s.rules, s.sb.Or(rules...))
+
+	ops := s.ops[len(s.ops)-2:]
+	s.ops = s.ops[:len(s.ops)-2]
+	s.ops = append(s.ops, &FilterOperation{
```

Comment:
do we have to add a rule for `OperatorNot` too? (`ExitNegated_search_expr` below)

Reply:
whoops, not sure how i missed this one. good catch!